### ISSUE TITLE
The dependency is ignored if a folder shares the same name as the package

### ISSUE THREAD

**Issue by ksesong** [kennethreitz/pipenv#1045]
This took me a few hours to find out, but when the current directory contains a folder which matches a dependency name, the dependency is simply ignored.

##### Describe your environment

1. OS Type: macOS 10.12.6
1. Python version: 2.7.10
1. Pipenv version: 8.3.2

##### Expected result

The package should be considered as a dependency.

##### Actual result

No dependency is found.
```
Locking [dev-packages] dependencies…
Using pip: -i https://pypi.python.org/simple

                          ROUND 1
Current constraints:

Finding the best candidates:

Finding secondary dependencies:
------------------------------------------------------------
Result of round 1: stable, done
```

##### Steps to replicate

Add to a directory the following `Pipfile`, and create an empty folder named `alembic`. Generate a `Pipfile.lock` with the `pipenv install` command, and you can see that the dependency has been ignored. 
```
[[source]]

url = "https://pypi.python.org/simple"
verify_ssl = true
name = "pypi"


[dev-packages]



[packages]

alembic = "==0.9.*"
```

An easy solution is to rename the `alembic` folder (which is what I've done), but as far as I know, this should not be necessary.

**Comment by techalchemy** [comment]
Hi @ksesong, thanks for the report.  This has to do as you guessed with having a local folder in the project root with the same name as a package you want to install. We are aware of this issue and have a fix for it in #958, and I agree you shouldn't have to rename the folder.  

I want to note that this doesn't prevent you from installing the package.  `Pipenv` will successfully install the package and its dependencies irrespective of the folder being present.  This only affects the generation of lockfiles.

In the meantime I am going to close this as we are tracking it in a few other places (#939, #936, #958 being the main ones) and this is a duplicate of #1025 and #949)

**Comment by techalchemy** [comment]
Resolved via #958
